dmrpp_generator/dmrpp.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import re
+
+import boto3
+import requests
+
+
+class DMRpp:
+    def __init__(self) -> None:
+        self.s3_client = boto3.client('s3')
+        self.session = requests.Session()
+
+    def get_https_file(self, url, host_path='/tmp'):
+        self.get_http_file(url, host_path)
+
+    def get_http_file(self, url, host_path="/tmp"):
+        """
+        Downloads a file from the provided url.
+        :param url: Location to find the file to download http or https
+        :param host_path: Where to store the downloaded file
+        """
+        filename = url.rsplit('/', 1)[-1]
```

Comment:
use os.path.basename(url) to get the name of the file

Reply:
The current implementation is more performant. See the following example:
```
if __name__ == "__main__":
    t = 'https://catalog.uah.edu/grad/colleges-departments/science/earth-system-science/earth-system-science.pdf'
    iterations = 1000000
    st = time.time()
    for _ in range(iterations):
        filename = t.rsplit('/', 1)[-1]
    et1 = time.time() - st

    st = time.time()
    for _ in range(iterations):
        os.path.basename(t)
    et2 = time.time() - st

    print(f'et1: {et1}, et2: {et2}')
    pass
```

Reply:
But the basepath is cleaner and more intuitive